engines/interpolantmc.cpp

```diff
@@ -120,26 +124,42 @@ bool InterpolantMC::step(int i)
   }
 
   Term bad_i = unroller_.at_time(bad_, i);
-  bad_disjuncts_ = solver_->make_term(Or, bad_disjuncts_, bad_i);
+  if (interp_props_ == InterpPropsEnum::ONLY_LAST || i == 1) {
+    // Note that by skipping the property at the first time frame,
+    // the computed interpolant might contain bad states
+    bad_disjuncts_ = bad_i;
+  } else if (interp_props_ == InterpPropsEnum::FIRST_AND_LAST) {
+    // Only take the first and last properties; the rest are skipped.
+    bad_disjuncts_ = solver_->make_term(Or, unroller_.at_time(bad_, 1), bad_i);
```

Comment:
unroller_.at_time(bad_, 0) ?

Reply:
1 is correct (see also https://github.com/stanford-centaur/pono/pull/381#discussion_r2125858854)